### PR TITLE
Implement Forecasting Model Training Pipeline and Data Management Framework

### LINKED ISSUE #11: bqa: Implements Training Pipeline
**Is your feature request related to a problem? Please describe.**
Currently, we lack a standardized and reproducible process for training our machine learning models. This leads to difficulties in tracking experiments, managing different datasets and model versions, and ensuring consistent results. Manually handling data loading, preprocessing, training, evaluation, and logging is time-consuming and prone to errors. Without a structured pipeline, collaborating on model development and deploying models reliably is challenging.

**Describe the solution you'd like**
We need to implement a comprehensive training pipeline that automates the end-to-end process of training machine learning models. This pipeline should include:
1.  **Data Loading:** Efficiently load training and validation datasets.
2.  **Data Processing:** Implement necessary preprocessing steps (e.g., feature scaling, encoding, handling missing values) as defined in the configuration.
3.  **Model Training:** Train an XGBoost model using the processed data.
4.  **Model Evaluation:** Evaluate the trained model using appropriate metrics on a validation set.
5.  **Result Visualization:** Generate visualizations of training progress and evaluation results.
6.  **Experiment Tracking:** Log training parameters, metrics, model artifacts, and visualizations using Weights & Biases (Wandb).
7.  **Configuration Management:** Manage all aspects of the pipeline (data paths, hyperparameters, preprocessing steps, etc.) using Hydra for clear, organized, and reproducible configurations.
8.  **Data Versioning:** Use DVC to version datasets and track dependencies between data and models, ensuring reproducibility of experiments.

The pipeline should be modular and easily extensible to incorporate different models or preprocessing steps in the future.

**Describe alternatives you've considered**
* **Manual Scripting:** Continuing with ad-hoc scripts for each step of the training process. This is the current state and is problematic due to lack of reproducibility, difficulty in tracking, and maintenance overhead.
* **Using a single, monolithic script:** Combining all steps into one large script. While slightly better than separate scripts, this approach lacks modularity, making it hard to test individual components, reuse code, and manage complexity.
* **Using other experiment tracking tools:** Considered tools like MLflow or TensorBoard. Wandb is preferred for its comprehensive features, ease of use, and strong community support.
* **Using other configuration management tools:** Considered tools like Argparse for simple cases or custom YAML loading. Hydra is preferred for its powerful features like hierarchical configuration, command-line overrides, and multi-run capabilities.
* **Using other data versioning tools:** Considered Git LFS for large files. DVC is preferred for its focus on data and model versioning, integration with Git, and pipeline definition capabilities.

**Additional context**
This training pipeline will be a core component of our machine learning workflow. Implementing this will significantly improve our ability to iterate on models, collaborate effectively, and deploy reliable models. The initial focus will be on integrating XGBoost, DVC, Wandb, and Hydra as specified. Future iterations may involve adding support for other model types or expanding the suite of preprocessing tools.

### PR DESCRIPTION
**Pull request by chogerlate**
# Implement Forecasting Model Training Pipeline and Data Management Framework

## Description
This PR implements a comprehensive forecasting model framework including data processing, model training, evaluation, and monitoring components. It introduces a standardized pipeline architecture with Hydra for configuration management and DVC for data versioning, specifically showcasing PM2.5 forecasting as an example use case.

## Related Issues
Closes #11 bqa: Implements Training Pipeline

## Changes Made
Added ModelTrainer class for structured model training workflows
Implemented data loading module with multi-source support
Created evaluation modules for forecasting performance metrics
Added monitoring and reporting capabilities for model performance
Integrated artifact management utilities for models, metrics, and datasets
Configured Hydra for flexible experiment management
Implemented DVC tracking for PM2.5 forecasting data
Updated project dependencies and configuration files
Added PR template for standardized submissions

## How to Test
Checkout this branch:
prerequisites
- uv
```
uv sync # Install dependencies
uv run dvc pull # Pull data from remote database
uv run src/baq/run.py # Run training pipeline 
```
image

## Checklist
[x] Code works as expected
[x] Self-review completed
[x] Tests added for core functionality
[x] Documentation updated for new modules
[x] DVC tracking implemented for example datasets
[x] Configuration files added for different experiment scenarios

## Notes
The forecasting pipeline currently supports univariate time series models with configurable horizon and feature engineering. Future PRs will expand this to multivariate models and add automated drift detection capabilities. The PM2.5 example dataset is included to demonstrate the complete workflow.

### CODE REVIEW

**Comment by kmpoppp** [comment]
LGTM!

**Comment by spppm** [comment]
LGTM
